### ISSUE TITLE
Handling exceptions

### ISSUE THREAD

**Issue by Schaussi** [cosullivan/SmtpServer#57]
Hi,
in rare cases it happens that unhandled exceptions (caused e.g. by myself) are thrown while I am processing emails, e.g. in the SaveAsync method.
I try to catch all possible exceptions, but for reasons I cannot really reproduce some of them are not caught and this results e.g. in a not responding SmtpServer.
Ttelnet'ing the server afterwards shows no welcome message etc. any more, but it also does not refuse the connection - so I would like to be able to restart the SmtpServer in such a case.
Because of the async type a try/catch block around server.StartAsync() would not help, of course.

Are there maybe some options etc. I can pass to the server before I start it?
In short, it would be great if I could simply restart the server on any exception (or catch and log every exception and let the server continue).

Thank you

**Comment by cosullivan** [comment]
Hi,

Are you using the latest version from Nuget? This really shouldn't happen in that if your code in the Save method does thrown an exception, it should be caught by the SmtpServer and the error is then propogated to the client. It shouldn't cause the server to stop accepting connections.

Do you have a sample you can share that reproduces the issue?

Thanks,
Cain.
